packages/clerk-js/src/ui/signUp/SignUpVerify.tsx

```diff
@@ -24,18 +24,17 @@ import {
 
 import { SignUpVerifyEmailAddressWithMagicLink } from './SignUpVerifyEmailAddressWithMagicLink';
 
-const emailLinkStrategy = 'email_link';
-
 function _SignUpVerifyEmailAddress(): JSX.Element {
-  const { authConfig } = useEnvironment();
-  const { firstFactors } = authConfig;
+  const { userSettings } = useEnvironment();
+  const { attributes } = userSettings;
 
   // TODO: SignUp should have a field similar to SignIn's supportedFirstFactors
   // listing the available strategies for this signUp
-  const emailLinkStrategyEnabled = firstFactors.includes(emailLinkStrategy);
+  const emailLinkStrategyEnabled =
+    attributes.email_address.first_factors.includes('email_link');
```

Comment:
❓ Does it make sense to also check if `attributes.email_address.used_for_first_factor` is true

```suggestion
    attributes.email_address.used_for_first_factor && attributes.email_address.first_factors.includes('email_link');
```

Reply:
I think it's enough since we only care about the `email_link` strategy, but @gkats might be able to confirm this